tests/integration/vxlan_test.py

```diff
@@ -0,0 +1,118 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+import time
+
+import pytest
+
+import libnmstate
+from libnmstate.error import NmstateVerificationError
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import InterfaceType
+
+from .testlib import assertlib
+from .testlib.vxlan import apply_state
+from .testlib.vxlan import vxlan_down
+from .testlib.vxlan import vxlan_interfaces
+from .testlib.vxlan import vxlan_up
+from .testlib.vxlan import vxlans_absent
+from .testlib.vxlan import vxlans_up
+
+VXLAN1_ID = 101
+VXLAN2_ID = 102
+
+
+def test_add_and_remove_vxlan(eth1_up):
+    ifname = eth1_up[Interface.KEY][0][Interface.NAME]
+    with vxlan_interfaces(
+        [(VXLAN1_ID, ifname, '192.168.100.1')]
```

Comment:
I suggest using regular arguments here or a data structure.
When defining an API between modules (this test module and its helper), it is preferred to avoid tuples/lists which include multiple types... you cannot track this easily and it is not structural visible to the reader.

Reply:
Going to refactor with OOP, this is growing out of proportions.